### ISSUE TITLE
Please fix hexo-deployer-heroku

### ISSUE THREAD

**Issue by yuanoOo** [hexojs/hexo-deployer-heroku#116]
## Check List

Please check followings before submitting a new issue.

- [x] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting)
- [x] I have already searched existing issues and they are not help  to me
- [x] I examined error or warning messages and it's difficult to solve
- [x] Using [the latest](https://github.com/hexojs/hexo/releases) version of Hexo (run `hexo version` to check)
- [x] Node.js is higher than [minimum required version](https://hexo.io/docs/#Minimum-required-Node-js-version)

## Actual behavior
i get something wrong , when i deploy hexo to heroku.

- the log:
```
INFO  Deploy done: git
INFO  Deploying: heroku
INFO  Copying files from public folder...
FATAL {
  err: TypeError [ERR_INVALID_ARG_TYPE]: The "mode" argument must be integer. Received an instance of Object
      at copyFile (node:fs:2791:10)
      at go$copyFile (/home/demo/hexo/hexo-blog/node_modules/graceful-fs/graceful-fs.js:181:14)
      at copyFile (/home/demo/hexo/hexo-blog/node_modules/graceful-fs/graceful-fs.js:178:12)
      at tryCatcher (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/util.js:16:23)
      at ret (eval at makeNodePromisifiedEval (/usr/local/lib/node_modules/hexo-cli/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:13:39)
      at /home/demo/hexo/hexo-blog/node_modules/hexo-deployer-heroku/node_modules/hexo-fs/lib/fs.js:181:12
      at tryCatcher (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/util.js:16:23)
      at Promise._settlePromiseFromHandler (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:547:31)
      at Promise._settlePromise (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:604:18)
      at Promise._settlePromise0 (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:649:10)
      at Promise._settlePromises (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:729:18)
      at Promise._fulfill (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:673:18)
      at Promise._resolveCallback (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:466:57)
      at Promise._settlePromiseFromHandler (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:559:17)
      at Promise._settlePromise (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:604:18)
      at Promise._settlePromise0 (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:649:10)
      at Promise._settlePromises (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/promise.js:725:18)
      at _drainQueueStep (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/async.js:93:12)
      at _drainQueue (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/async.js:86:9)
      at Async._drainQueues (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/async.js:102:5)
      at Immediate.Async.drainQueues [as _onImmediate] (/home/demo/hexo/hexo-blog/node_modules/bluebird/js/release/async.js:15:14)
      at processImmediate (node:internal/timers:466:21) {
    code: 'ERR_INVALID_ARG_TYPE'
  }
} Something's wrong. Maybe you can find the solution here: %s https://hexo.io/docs/troubleshooting.html
```
- my env:
```
hexo: 6.2.0
hexo-cli: 4.3.0
os: linux 5.10.16.3-microsoft-standard-WSL2 Ubuntu 20.04 LTS (Focal Fossa)
node: 16.15.1
v8: 9.4.146.24-node.21
uv: 1.43.0
zlib: 1.2.11
brotli: 1.0.9
ares: 1.18.1
modules: 93
nghttp2: 1.47.0
napi: 8
llhttp: 6.0.4
openssl: 1.1.1o+quic
cldr: 40.0
icu: 70.1
tz: 2021a3
unicode: 14.0
ngtcp2: 0.1.0-DEV
nghttp3: 0.1.0-DEV
```
- my package.json
```
  1 {
  2   "name": "hexo-site",
  3   "version": "0.0.0",
  4   "private": true,
  5   "scripts": {
  6     "build": "hexo generate",
  7     "clean": "hexo clean",
  8     "deploy": "hexo deploy",
  9     "server": "hexo server"
 10   },
 11   "hexo": {
 12     "version": "6.2.0"
 13   },
 14   "dependencies": {
 15     "hexo": "^6.2.0",
 16     "hexo-deployer-git": "^3.0.0",
 17     "hexo-deployer-heroku": "^0.1.2",
 18     "hexo-generator-archive": "^1.0.0",
 19     "hexo-generator-category": "^1.0.0",
 20     "hexo-generator-index": "^2.0.0",
 21     "hexo-generator-tag": "^1.0.0",
 22     "hexo-renderer-ejs": "^2.0.0",
 23     "hexo-renderer-marked": "^5.0.0",
 24     "hexo-renderer-stylus": "^2.1.0",
 25     "hexo-server": "^3.0.0",
 26     "hexo-theme-fluid": "^1.9.2",
 27     "hexo-theme-landscape": "^0.0.3"
 28   }
 29 }
```

## How to reproduce?

* Step1
* Step2
* etc...

## Environment & Settings

**Node.js & npm version(`node -v && npm -v`)**

```
```

**Your site `_config.yml`** (Optional)

```yaml
```

**Hexo and Plugin version(`npm ls --depth 0`)**

```
```

**Your package.json `package.json`**

```
```

**Comment by tomizechsterson** [comment]
I'm getting the same results when trying `hexo deploy` with Heroku

**Comment by tomizechsterson** [comment]
Just opened a PR on `hexo-fs` that gets past this error: https://github.com/hexojs/hexo-fs/pull/100 . Unfortunately, it failed for me on the Heroku side after this, but at least there's a potential to get past this particular error

**Comment by awongCM** [comment]
Hi,

Has there been any progress in getting the deploy plugin fixed so the `hexo deploy` still works correctly as before?  It used to work for me back in May 2022 when I last published my personal blog.... But now I tried doing this tonight, it not longer works as before.  I encountered

`ERROR: Application not supported by 'heroku/nodejs' buildpack `

instead of ERR_INVALID_ARG_TYPE.  

I worked around ERR_INVALID_ARG_TYPE by pinning down the `node` and `npm` versions in package.json, but ended up with a buildpack error instead.

Not sure why this plugin failed all of a sudden..

But out of desperation, I decided to follow the official Heroku docs to add 'Heroku git URL', and it works fine for me straight after.

**Comment by awongCM** [comment]
> Hi,
> 
> Has there been any progress in getting the deploy plugin fixed so the `hexo deploy` still works correctly as before? It used to work for me back in May 2022 when I last published my personal blog.... But now I tried doing this tonight, it not longer works as before. I encountered
> 
> `ERROR: Application not supported by 'heroku/nodejs' buildpack `
> 
> instead of ERR_INVALID_ARG_TYPE.
> 
> I worked around ERR_INVALID_ARG_TYPE by pinning down the `node` and `npm` versions in package.json, but ended up with a buildpack error instead.
> 
> Not sure why this plugin failed all of a sudden..
> 
> But out of desperation, I decided to follow the official Heroku docs to add 'Heroku git URL', and it works fine for me straight after.

Another possible reason I suspect is that the due to the recent changes by Heroku towards end of 2022, they have changed the structure of the buildpacks requirements such that it needs to detect the `package.json` to be available at the root of the `.deploy_heroku` (not the project hexo root folder) before you can deploy anything successfully..

**Comment by stevenjoezhang** [comment]
We may need to release a new version of the plugin, to update the version of hexo-fs dependency. CC @hexojs/core

**Comment by Ilya122** [comment]
Hi
I forked this repo, linked via ``npm link`` my local folder to the project I build. 

I added options in the ``fs.copyFile`` function and it seemed to work.
https://github.com/hexojs/hexo-deployer-heroku/compare/master...Ilya122:hexo-deployer-heroku:master

**Comment by Ilya122** [comment]
> > Hi I forked this repo, linked via `npm link` my local folder to the project I build.
> > I added options in the `fs.copyFile` function and it seemed to work. [master...Ilya122:hexo-deployer-heroku:master](https://github.com/hexojs/hexo-deployer-heroku/compare/master...Ilya122:hexo-deployer-heroku:master)
> 
> The latest version `hexo-fs` already provides the default option for `copyDir`:
> 
> https://github.com/hexojs/hexo-fs/blob/f1b49b28c28d1fee4024c84a08a6b485c8253dec/lib/fs.ts#L160
> 
> So `hexo-deployer-heroku` will only need to update `hexo-fs` to work.

Seems so.
+ it needs all relevant node packages also updated.

**Comment by yoshinorin** [comment]
@yuanoOo @tomizechsterson @Ilya122 @awongCM 

https://github.com/hexojs/hexo-deployer-heroku/releases/tag/v1.0.0
https://www.npmjs.com/package/hexo-deployer-heroku/v/1.0.0?activeTab=versions

Just now we released `v1.0.0`.
Thank you for your reporting :)
